Sources/_CryptoExtras/SPX/SPX_boring.swift

```diff
@@ -0,0 +1,176 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftCrypto open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftCrypto project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.md for the list of SwiftCrypto project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+@_implementationOnly import CCryptoBoringSSL
+@_implementationOnly import CCryptoBoringSSLShims
+import Crypto
+@_implementationOnly import CryptoBoringWrapper
+import Foundation
+
+#if swift(>=5.8)
+@_documentation(visibility: public)
+public enum SPX {}
+#else
+public enum SPX {}
+#endif
+
+extension SPX {
+    public struct PrivateKey: Sendable {
+        private let pointer: UnsafeMutablePointer<UInt8>
+        
+        public init() {
+            self.pointer = UnsafeMutablePointer<UInt8>.allocate(capacity: 64)
+            CCryptoBoringSSL_spx_generate_key(UnsafeMutablePointer<UInt8>.allocate(capacity: 32), self.pointer)
+        }
+        
+        public init(from seed: some DataProtocol) throws {
+            guard seed.count >= (3 * 16) else {
+                throw CryptoKitError.incorrectKeySize
+            }
+            let seedPtr = UnsafeMutablePointer<UInt8>.allocate(capacity: 3 * 16)
+            seedPtr.initialize(from: seed.regions.flatMap { $0 }, count: 3 * 16)
+            self.pointer = UnsafeMutablePointer<UInt8>.allocate(capacity: 64)
+            CCryptoBoringSSL_spx_generate_key_from_seed(UnsafeMutablePointer<UInt8>.allocate(capacity: 32), self.pointer, seedPtr)
+        }
+
+        public init<Bytes: DataProtocol>(derRepresentation: Bytes) throws {
+            guard derRepresentation.count == 64 else {
```

Comment:
`public static let derLength = 64` and use that in the three places of this initializer?